### ISSUE TITLE
 pkg/sql/logictest/tests/local: `TestLogic_pg_catalog` is flakey

### ISSUE THREAD

**Issue by fantapop** [cockroachdb/cockroach#86960]
**Describe the problem**

<img width="600" alt="Screen Shot 2022-08-26 at 11 48 08 AM" src="https://user-images.githubusercontent.com/6658984/186971828-18c2f763-1e6d-4b25-9050-4e9b0798a923.png">

Fails with: 
```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1316/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/pg_catalog:4657: SELECT p.oid, p.proname, t.typinput
        FROM pg_proc p
        JOIN pg_type t ON t.typinput = p.oid
        WHERE t.typname = '_int4'
        expected:
            2013  array_in  array_in
            
        but found (query options: "") :
            2014  array_in  array_in
```

every so often

Jira issue: CRDB-19059

**Comment by fantapop** [comment]
I see this is a generated test so it will be tricky to skip it.  Hoping someone can take a look at it instead.

**Comment by cucaroach** [comment]
Adding to SQL experience triage, I looked at the test and don't have the foggiest clue what's going on.

**Comment by rickystewart** [comment]
> I see this is a generated test so it will be tricky to skip it. Hoping someone can take a look at it instead.

I can help with adding code to `generate-logictest` to skip certain tests, should be relatively easy if that's the route we want to go down.

**Comment by cucaroach** [comment]
I poked around on teamcity but can't find any instances of this failing.  Where are we seeing this?

**Comment by rafiss** [comment]
I think this is now addressed by 5ee5d85fbbd64db0a14caea366f2911c2460e4ac
